File: Speedy-Go-backend/src/main/java/com/ski/speedygobackend/Entity/OfferManagement/Store.java
package com.ski.speedygobackend.Entity.OfferManagement;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.ski.speedygobackend.Entity.UserManagement.User;
import com.ski.speedygobackend.Enum.StoreStatus;
import com.ski.speedygobackend.Enum.StoreType;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import java.io.Serializable;
import java.time.LocalTime;
import java.util.Set;


@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level=AccessLevel.PRIVATE)
@Entity
public class Store  implements Serializable {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  Long storeID;
  String name;
  String opening;
  String closing;
  String logo;
  String website;
  String image;
  String address;
  String city;
  String location;
  String description;
  String phone;
  String email;

  @Enumerated(EnumType.STRING)
  @JsonProperty("storeType")
  private StoreType storeType;

  @Enumerated(EnumType.STRING)
  @JsonProperty("storeStatus")
  private StoreStatus storeStatus;


  @ManyToOne
  private User user;
  @OneToMany(mappedBy = "store")
  private Set<Offres> offres;
}

File: Speedy-Go-frontend/src/app/FrontOffices/modules/store/store/model/store.ts
// src/app/models/store.model.ts
export enum StoreType {
    SHOP = 'SHOP',
    RESORT = 'RESORT',
    CAFE = 'CAFE',
    ELECTRONICS = 'ELECTRONICS',
    CLUB = 'CLUB',
    OTHERS = 'OTHERS'
  }
  
  export enum StoreStatus {
    OPEN = 'OPEN',
    CLOSED = 'CLOSED',
    SUSPENDED = 'SUSPENDED'
  }
  
  export interface Store {
    storeID?: number;
    name: string;
    opening?: string;
    closing?: string;
    logo: string;
    website: string;
    image: string;
    address: string;
    city: string;
    location: string;
    description: string;
    phone?: string;
    email: string;
    storeType: StoreType;
    storeStatus: StoreStatus;
    // References to related entities
    user?: any;
    offres?: any[];
  }